src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write your code here
+const promise1 = new Promise(function(resolve, reject) {
+  const logo = document.querySelector('.logo');
+
+  logo.addEventListener('click', e => {
+    if (!e.target.closest('.logo')) {
+      return;
+    }
+    resolve();
+  });
+});
+
+promise1
+  .then(() => {
+    const body = document.querySelector('body');
+
+    body.insertAdjacentHTML('beforeend', `
+      <div class="message">Promise was resolved!</div>
+    `);
```

Comment:
In both cases you do almost the same
You can move this logic to a separate function.